libdnf/utils/fs/temp.cpp

```diff
@@ -92,39 +92,33 @@ TempFile::~TempFile() {
 }
 
 
-FILE * TempFile::fdopen(const char * mode) {
-    libdnf_assert(fd != -1, "Cannot fdopen a TempFile that has been closed or released");
+File & TempFile::open_as_file(const char * mode) {
+    libdnf_assert(fd != -1, "Cannot open as file TempFile that has been closed or released");
 
-    file = ::fdopen(fd, mode);
-    if (file == nullptr) {
-        throw std::filesystem::filesystem_error(
-            "cannot open temporary file", path, std::error_code(errno, std::system_category()));
-    }
-
-    return file;
+    file.emplace(fd, path, mode);
```

Comment:
What if someone calls `open_as_file` twice. filedescriptor will be closed (invalidated) and then invalid will be used for reopening. That will not turn out well.